astropy/units/physical.py

```diff
@@ -314,6 +314,13 @@ def __iter__(self):
         yield from self._physical_type_list
 
     def __getattr__(self, attr):
+        # TODO: remove this whole method when accessing str attributes from
+        # physical types is no longer supported
+
+        # short circuit attribute accessed in __str__ to prevent recursion
+        if attr == '_physical_type_list':
+            super().__getattribute__(attr)
```

Comment:
Interesting!  I was pretty confused by this error, but I'm glad this fixes it.  Kind thanks!